components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -0,0 +1,380 @@
+import { findComposedAncestor, isComposedAncestor } from '../../helpers/dom.js';
+import { getNextFocusable, getPreviousFocusable } from '../../helpers/focus.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+export const HierarchicalViewMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			shown: { type: Boolean, reflect: true },
+			childView: { type: Boolean, reflect: true, attribute: 'child-view' },
+			isAttached: { type: Boolean, reflect: true, attribute: 'is-attached' },
+			isHierarchicalView: { type: Boolean, reflect: true, attribute: 'is-hierarchical-view' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		this.isHierarchicalView = true;
+		this.__nativeFocus = null;
+		this.__focusPrevious = false;
+		this.__resizeObserver = null;
+	}
+
+	firstUpdated() {
+		super.firstUpdated();
+		this.addEventListener('keyup', this.__onKeyUp);
+		this.addEventListener('d2l-hierarchical-view-hide-start', this.__onHideStart);
+		this.addEventListener('d2l-hierarchical-view-show-start', this.__onShowStart);
+		this.addEventListener('d2l-hierarchical-view-resize', this.__onViewResize);
+		this.__nativeFocus = document.createElement('div').focus;
+		this.__onWindowResize = this.__onWindowResize.bind(this);
+
+		const parentView = findComposedAncestor(
+			this.parentNode,
+			(node) => { return node.isHierarchicalView; }
+		);
+
+		if (parentView) {
+			this.childView = true;
+		}
+
+		if (!this.childView) {
+			this.addEventListener('focus', this.__focusCapture, true);
+			this.addEventListener('focusout', this.__focusOutCapture, true);
+			window.addEventListener('resize', this.__onWindowResize);
```

Comment:
The `resize` event handler is wired up to the `window`, and there is no call to remove the handler, so it (and the element it references) would not get cleaned up until the user navigates away or closes their browser.  

The `firstUpdated` is preferred when wiring up handlers to elements that are in the component's subtree (wire-up will be deferred until the shadow-DOM has been created; automatically cleaned up when element is garbage collected), but when wiring up to elements outside, we need to take more control to explicitly add/remove the handler, most likely in `connectedCallback` and `disconnectedCallback`. 

Hopefully we can know if it's a childView in the `connectedCallback`, because we only want the `resize` handler wired up for the root view, since the root-view controls dimensions.